crypto/bio/bio_lib.c

```diff
@@ -796,31 +796,32 @@ static int bio_wait(BIO *bio, time_t max_time, unsigned int milliseconds)
         return 1;
 
 #ifndef OPENSSL_NO_SOCK
-    if (BIO_get_fd(bio, &fd) > 0)
+    if (BIO_get_fd(bio, &fd) > 0 && fd < FD_SETSIZE)
         return BIO_socket_wait(fd, BIO_should_read(bio), max_time);
 #endif
-    if (milliseconds > 1000) {
+    if (fallback_millisec > 1000) {
         long sec_diff = (long)(max_time - time(NULL)); /* might overflow */
 
         if (sec_diff <= 0)
             return 0; /* timeout */
-        if ((unsigned long)sec_diff < milliseconds / 1000)
-            milliseconds = (unsigned long)sec_diff * 1000;
+        if ((unsigned long)sec_diff < fallback_millisec / 1000)
+            fallback_millisec = (unsigned long)sec_diff * 1000;
```

Comment:
why are you casting to unsigned long, when fallback_millisec is unsigned int ?

Reply:
This is a leftover from an earlier version where the type of `(fallback_)millisec` was `long` and not needed any more. Removed.